lib/browse_everything/driver/file_system.rb

```diff
@@ -36,18 +44,32 @@ def details(path, display = nil)
         return nil unless File.exist?(path)
         info = File::Stat.new(path)
         BrowseEverything::FileEntry.new(
-          Pathname.new(File.expand_path(path)).relative_path_from(Pathname.new(config[:home])),
+          make_pathname(path),
           [key, path].join(':'),
-          display || File.basename(path),
+          make_display(path, display),
```

Comment:
Rather than this, how about changing the signature to `details(path, display = File.basename(path))`? Then you could just use `display` here.